CONTRIBUTING.md

```diff
@@ -59,3 +59,92 @@
 
 ✅ _px-to-rem.scss          # Correct
 ```
+
+## Semantic Commits: A Complete Guide 📝
+
+## What are semantic commits? 🤔
+
+Semantic commits are a way of writing commit messages that follow a standardized structure, making them more readable and understandable for both humans and machines.
+
+## Benefits of semantic commits 🚀
+
+- Enhance communication and understanding of project history.
+- Facilitate task automation, such as generating changelogs or version management.
+- Enable more precise tracking of project progress.
+
+## Structure of a semantic commit 🏗️
+
+A semantic commit consists of three main parts:
+
+1. **Commit type**: Indicates the type of change made. Common commit types include:
+   - :sparkles: feat: Introduces a new feature.
+   - :bug: fix: Fixes a bug.
+   - :wrench: chore: Project maintenance tasks, such as dependency updates.
+   - :page_facing_up: docs: Documentation changes.
+   - :art: style: Code style changes, such as indentation or formatting.
+   - :white_check_mark: test: Adds or modifies tests.
+   - :recycle: refactor: Code structure changes without modifying its functionality.
+
+2. **Scope (optional)**: Specifies the area of the project affected by the change. It is written in parentheses after the commit type.
+
+3. **Description**: A brief description of the change made. It should be clear, concise, and understandable to anyone reading the project history.
+
+## Example of a semantic commit 💡
+
+```bash
+feat(parser): add ability to parse arrays
+```
+```bash
+fix(validation): Fixed validation issue with user input
+```
+```bash
+chore(dependencies): Updated dependencies to latest versions
+```
+```bash
+docs(readme): Updated README.md with new instructions
+```
+```bash
+style(css): Adjusted indentation in stylesheet for consistency
+```
+```bash
+test(unit): Added unit tests for authentication module
+```
+```bash
+refactor(api): Restructured API endpoints for improved readability
+```
+```bash
+build: Updated build configuration for better performance
+```
+```bash
+ci: Updated continuous integration configuration for faster builds
+```
+```bash
+perf(database): Optimized database queries for faster response times
+```
+```bash
+revert: Reverted previous commit that introduced bug in authentication
+```
+```bash
+security(auth): Fixed security vulnerability in authentication process
+```
```

Comment:
```suggestion
```bash
feat(parser): add ability to parse arrays
fix(validation): Fixed validation issue with user input
chore(dependencies): Updated dependencies to latest versions
docs(readme): Updated README.md with new instructions
style(css): Adjusted indentation in stylesheet for consistency
test(unit): Added unit tests for authentication module
refactor(api): Restructured API endpoints for improved readability
build: Updated build configuration for better performance
ci: Updated continuous integration configuration for faster builds
perf(database): Optimized database queries for faster response times
revert: Reverted previous commit that introduced bug in authentication
security(auth): Fixed security vulnerability in authentication process
```